apps/transport/lib/transport_web/views/dataset_view.ex

```diff
@@ -291,13 +291,13 @@ defmodule TransportWeb.DatasetView do
       when type == "carsharing-areas" or type == "private-parking" or type == "charging-stations" do
     resources
     |> Enum.filter(fn r -> r.format == "csv" end)
-    |> Enum.max_by(fn r -> r.last_update end)
+    |> Enum.max_by(fn r -> r.last_update end, fn -> nil end)
```

Comment:
Si on est bien sur [cette fonction](https://hexdocs.pm/elixir/Enum.html#max_by/4), le `fn -> nil end` fait référence si je ne me trompe pas à `sorter`, plutôt que `empty_fallback` ?

Ça m'interpelle mais j'imagine que tu as testé en vrai, est-ce que ce code ne remplace pas la fonction de sort plutôt que le falback?

Reply:
Un petit test pourrait aider là dessus, mais je ne sais pas encore si c'est simple à faire ni super pertinent, et puis on souhaite aussi un hot-fix !

Reply:
en 1.8 c'est pas la même fonction :stuck_out_tongue_winking_eye: https://hexdocs.pm/elixir/1.8/Enum.html#max_by/3
(mais oui faut faire attention à l'upgrade...)

Reply:
Correction: la documentation que j'ai linké est pour Elixir 1.11.2, alors qu'on est sur 1.8 (#1336).

La bonne documentation est [ici](https://hexdocs.pm/elixir/1.8.1/Enum.html#max_by/3).

Reply:
@antoine-de je suggère de mettre un TODO avec un lien vers ce point dans le code !

Reply:
Pas besoin de TODO, car la rétro-compatibilité semble assurée.

Voici en Elixir 1.10.3:

```
Enum.max_by([], fn r -> -r  end, fn -> raise(Enum.EmptyError) end)
```

lève bien une erreur.

Par ailleurs ce point n'apparaît pas sur https://hexdocs.pm/elixir/compatibility-and-deprecations.html, donc ça semble avoir été géré en mode "soft deprecation", même si je n'ai pas été voir le détail du code.

Mais c'est rassurant !

Reply:
pour le test carrément. aprés l’intérêt d'un doc test ici est toujours limité vu qu'on doit "mocker" les datasets en entrée. j'en tente et on voit ce que ca donne